submarine-sdk/pysubmarine/setup.py

```diff
@@ -28,41 +28,41 @@
     packages=find_packages(exclude=["tests", "tests.*"]),
     package_data={"submarine.cli.config": ["cli_config.yaml"]},
     install_requires=[
-        "six>=1.10.0",
-        "numpy==1.19.2",
+        "numpy",
         "pandas",
         "sqlalchemy>=1.4.0",
         "sqlparse",
         "pymysql",
-        "requests==2.26.0",
+        "requests",
         "urllib3>=1.15.1",
         "certifi>=14.05.14",
         "python-dateutil>=2.5.3",
         "pyarrow>=6.0.1",
         "boto3>=1.17.58",
-        "click==8.0.3",
-        "rich==10.15.2",
-        "dacite==1.6.0",
-        "dataclasses>=0.6",
-        "pyaml==21.10.1",
+        "click>=8.1.0",
+        "rich",
+        "dacite",
+        "pyaml",
     ],
     extras_require={
-        "tf": ["tensorflow==1.15.0", "protobuf<=3.20.1"],
+        "tf": ["tensorflow==1.15.5", "numpy>=1.16.0,<1.19.0", "protobuf>=3.6.1,<=3.20.1"],
         "tf2": [
-            "tensorflow==2.6.0",
+            "tensorflow>=2.6.0,<2.10.0",
+            "numpy>=1.14.5",
+            "keras>=2.6.0",
+            "protobuf>=3.9.2,<3.20",
```

Comment:
`protobuf` in tf2 should also be limited to 3.20.1 ?
I don't know what the latest versions of all these dependencies are yet, and it may take me some time to confirm whether there is some potential impact of the interdependencies between these versions. Or if you have time on your side could you run a test locally? 
What do you think? @pingsutw @KUAN-HSUN-LI

Reply:
The protobuf for Tensorflow 2.9.2 is limited to 3.20.
<https://github.com/tensorflow/tensorflow/blob/v2.9.2/tensorflow/tools/pip_package/setup.py#L97>

I will later list the protobuf version constraints for each Tensorflow2 version.

Reply:
This [setup.py](https://github.com/tensorflow/tensorflow/blob/v2.9.2/tensorflow/tools/pip_package/setup.py#L90-L97) shows that protobuf 3.20 encounters errors on Windows.
Currently, all the TF2 patch versions are set to `protobuf >= 3.9.2, < 3.20`.

```python
# TODO(b/182876485): Protobuf 3.20 results in linker errors on Windows
# Protobuf 4.0 is binary incompatible with what C++ TF uses.
# We need ~1 quarter to update properly.
# See also: https://github.com/tensorflow/tensorflow/issues/53234
# See also: https://github.com/protocolbuffers/protobuf/issues/9954
# See also: https://github.com/tensorflow/tensorflow/issues/56077
# This is a temporary patch for now, to patch previous TF releases.
'protobuf >= 3.9.2, < 3.20',
```

This [table](https://gist.github.com/huang06/811270ba210f1a0f858e7f1d9138d6fc#file-ml-package-deps-md) lists detailed dependencies for each version.

Reply:
Should we remain `protobuf>=3.9.2,<3.20` or relax the version constraint (e.g. `protobuf>=3.9.2`) for future Tensorflow patches?